pkg/client/simple/vfsclientset/clientset.go

```diff
@@ -84,6 +93,73 @@ func (c *VFSClientset) GetFederation(name string) (*kops.Federation, error) {
 	return c.federations().Get(name, metav1.GetOptions{})
 }
 
+func (c *VFSClientset) SecretStore(cluster *kops.Cluster) (fi.SecretStore, error) {
+	configBase, err := registry.ConfigBase(cluster)
+	if err != nil {
+		return nil, err
+	}
+	basedir := configBase.Join("secrets")
+	return secrets.NewVFSSecretStore(basedir), nil
+}
+
+func (c *VFSClientset) KeyStore(cluster *kops.Cluster) (fi.CAStore, error) {
+	configBase, err := registry.ConfigBase(cluster)
+	if err != nil {
+		return nil, err
+	}
+	basedir := configBase.Join("pki")
+	return fi.NewVFSCAStore(basedir), nil
+}
+
+func DeleteAllClusterState(basePath vfs.Path) error {
```

Comment:
I will need to test this in a subfolder state store without list perms.  I will open an issue for me.

Reply:
This function was moved from kops/registry